### PR TITLE
Adds save test for saveLastRetiredFederationP2SHScript

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds save test for saveLastRetiredFederationP2SHScript

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -1186,6 +1186,31 @@ void saveNextFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNotNu
 
     }
 
+    @Test
+    void save_saveLastRetiredFederationP2SHScript_afterRSKIP186_getsValueFromStorage() {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1190]: remove the `save_` prefix

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -1186,6 +1186,31 @@ void saveNextFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNotNu
...
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        Script expectedScript = new P2shErpFederationBuilder().build().getP2SHScript();
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1202]: isn't this line be on the _Arrange_ part?

**Review by julia-zack — APPROVED** [review]
lgtm, left two minor comments

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (wilmerrootstock) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1190]: `save` is the method name we are testing. We might leave the `save_` prefix and remove it from what comes after, thus making clear what method we are testing. E.g.:

`save_LastRetiredFederationP2SHScript_afterRSKIP186_getsValueFromStorage()
`
WDYT guys? @julia-zack @jeremy-then

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -1186,6 +1186,31 @@ void saveNextFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNotNu
...
+        federationStorageProvider.setLastRetiredFederationP2SHScript(expectedScript);
```
Comment (wilmerrootstock) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1203]: This line should be on the `arrange` part, as well as, the previous code line.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1190]: if he removes the `save_` prefix, the test name would be `saveLastRetiredFederationP2SHScript_afterRSKIP186_getsValueFromStorage`.

Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1223]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1224]: Updated.

**Review by wilmerrootstock — APPROVED** [review]
LGTM!

**Pull request closed without merging**